test/expected/bgw_scheduler.out

```diff
@@ -0,0 +1,248 @@
+\c single_2 :ROLE_SUPERUSER
+/* 
+ * Note on testing this: we had to add a short pg_sleep in a few places
```

Comment:
Yeah. I'm a bit worried about this as well. Unfortunately, I can't figure a way around it as of now. But would appreciate advice/workarounds if you have any. The basic gist is we are already waiting on the worker starting up, but it has to actually have fully started up and reach a certain point inside the entrypoint function before we get into pg_stat_activity and set our application name. It is basically the first thing we do when we enter the function, but it still takes some minor amount of time. I haven't seen tests flaking when I run them on my machine, but I would want to see if a) we don't see flaky tests if we run it on a number of machines and b) we can think of some better way of figuring this out. I can't think of one without adding a fair amount of extra overhead in terms of code/complexity (some sort of signaling mechanism back to the launcher). Now it might be we want that complexity for other reasons. But I wouldn't necessarily want to add it in just because of the possibility of a flaky test.

Reply:
A common pattern for solving sleep issues is to use latches or locks in some wrapper function that then calls the real function. This often requires that your main code is written to be amenable to testing, which is a bit more work. However, the upside is that that work typically pays off because your code is then more flexible, modular, and testable.

In this particular case, could you pass a `dsm_handle` in `bgw_extra` and use as a startup ACK that you send in the worker once it has properly registered its name? Then you can wait on that ACK in the after WaitForBackgroundWorkerStartup. Seems you've done most of the work for that already,.

Reply:
I'm not sure that the extra complexity involved is worth it for the MVP, but I think perhaps it would be good for you, me and Mat to look it over and see how that would get implemented.